tests/php/TestCase.php

```diff
@@ -0,0 +1,99 @@
+<?php # -*- coding: utf-8 -*-
+
+namespace Mollie\WooCommerce\Tests;
+
+use function Brain\Monkey\setUp;
+use function Brain\Monkey\tearDown;
+use Mockery;
+use PHPUnit\Framework\TestCase as PhpUniTestCase;
+use PHPUnit_Framework_MockObject_MockBuilder;
+use ReflectionException;
+use ReflectionMethod;
+
+/**
+ * Class Testcase
+ */
+class TestCase extends PhpUniTestCase
+{
+    /**
+     * Sets up the fixture, for example, open a network connection.
+     * This method is called before a test is executed.
+     */
+    protected function setUp()
+    {
+        parent::setUp();
+        setUp();
+    }
+
+    /**
+     * Tears down the fixture, for example, close a network connection.
+     * This method is called after a test is executed.
+     */
+    protected function tearDown()
+    {
+        parent::tearDown();
+        Mockery::close();
+        tearDown();
+    }
+
+    /**
+     * Build the Testee Mock Object
+     *
+     * Basic configuration available for all of the testee objects, call `getMock` to get the mock.
+     *
+     * @param string $className
+     * @param array $constructorArguments
+     * @param array $methods
+     * @param string $sutMethod
+     * @return PHPUnit_Framework_MockObject_MockBuilder
+     */
+    protected function buildTesteeMock(
+        $className,
+        $constructorArguments,
+        $methods,
+        $sutMethod
+    ) {
+
+        $testee = $this->getMockBuilder($className);
+        $constructorArguments
+            ? $testee->setConstructorArgs($constructorArguments)
+            : $testee->disableOriginalConstructor();
+
+        $methods and $testee->setMethods($methods);
+        $sutMethod and $testee->setMethodsExcept([$sutMethod]);
+
+        return $testee;
+    }
+
+    /**
+     * Retrieve a Testee Mock to Test Protected Methods
```

Comment:
I know but I never had the need to use Andrew since I create that simple methods and them are working fine so :D

Reply:
They work fine, sure. But they make the test harder to understand. Returning and using a mock and an invocable is not nearly as straight-forward as invoking a method on an object.

Reply:
That's not a problem, isn't more hard to understand than learn a new package such as Andrew or xpmock. I use it because make more simple to create the tests mock for methods.

Reply:
I would disagree. There's nothing to understand about xpmock's reflections that is different for consumers to Andrew. And the only thing you need to know about consuming them is that you need to invoke protected properties and methods - just like you normally would. That's not something I could say about your method, which makes things more simpler than _not_ having one at all, but far less simple than Andrew or xpmock. But if you say that test readability and clarity is not a problem, then I'm OK with that.

Reply:
> There's nothing to understand about xpmock's reflections that is different for consumers to Andrew.
Not talking about the differences between them, I'm talking about the unnecessary need of introducing a new library for a thing that can be accomplished by one single method.

> And the only thing you need to know about consuming them is that you need to invoke protected properties and methods 

Not with the method I implemented, you just ask to build the necessary stuffs to make a test for a method, you don't care about visibility which is nice because you don't need to know it in test context. Plus you don't depend on additional library for a simple task like this and give you the free to manipulate the code as per your needs, something you cannot do with third party libraries. 

In testing I prefer to have more freedom that depends on other libraries when I can just solve a problem with a small function or method.

> That's not something I could say about your method, which makes things more simpler than not having one at all, but far less simple than Andrew or xpmock. But if you say that test readability and clarity is not a problem, then I'm OK with that.

My method doesn't make the test less readable than having a third party library only because you know those libraries. During test you just want to setup the SUT to have it testable, you don't care about which library will be used under the hood and that doesn't make the code less clear. You don't care about visibility because you want to do unit test.

Reply:
During a video call, it became apparent that the disagreement was due to a misunderstanding. The new method should be implemented now.

Reply:
I would create the proxy from the mock in the test explicitly, rather than return the proxy instead of the mock. But this will also do.